docs/01-app/05-api-reference/03-file-conventions/layout.mdx

```diff
@@ -119,78 +119,466 @@ export default function RootLayout({ children }) {
 
-This design also enforces consistent and predictable behavior for layouts across different pages, which simplifies development and debugging.
+To access the request object, you can use [`headers`](/docs/app/api-reference/functions/headers) and [`cookies`](/docs/app/api-reference/functions/cookies) APIs in [Server Components](/docs/app/getting-started/server-and-client-components) and Functions.
 
-### Layouts do not receive `searchParams`
+```tsx filename="app/shop/layout.tsx" switcher
+import { cookies } from 'next/headers'
 
-Unlike [Pages](/docs/app/api-reference/file-conventions/page), Layout components **do not** receive the `searchParams` prop. This is because a shared layout is [not re-rendered during navigation](/docs/app/building-your-application/routing/linking-and-navigating#4-partial-rendering) which could lead to stale `searchParams` between navigations.
+export default async function Layout({ children }) {
+  const cookieStore = await cookies()
+  const theme = cookieStore.get('theme')
+  return '...'
+}
+```
 
-When using client-side navigation, Next.js automatically only renders the part of the page below the common layout between two routes.
+```jsx filename="app/shop/layout.js" switcher
+import { cookies } from 'next/headers'
 
-For example, in the following directory structure, `dashboard/layout.tsx` is the common layout for both `/dashboard/settings` and `/dashboard/analytics`:
+export default async function Layout({ children }) {
+  const cookieStore = await cookies()
+  const theme = cookieStore.get('theme')
+  return '...'
+}
+```
 
-<Image
-  alt="File structure showing a dashboard folder nesting a layout.tsx file, and settings and analytics folders with their own pages"
-  srcLight="/docs/light/shared-dashboard-layout.png"
-  srcDark="/docs/dark/shared-dashboard-layout.png"
-  width="1600"
-  height="687"
-/>
+### Query params
 
-When navigating from `/dashboard/settings` to `/dashboard/analytics`, `page.tsx` in `/dashboard/analytics` will rerender on the server, while `dashboard/layout.tsx` will **not** rerender because it's a common UI shared between the two routes.
+Layouts do not rerender on navigation, so they cannot access search params which would otherwise become stale.
 
-This performance optimization allows navigation between pages that share a layout to be quicker as only the data fetching and rendering for the page has to run, instead of the entire route that could include shared layouts that fetch their own data.
+To access updated query parameters, you can use the Page[`searchParams`](/docs/app/api-reference/file-conventions/page#searchparams-optional) prop, or read them inside a Client Component using the [`useSearchParams`](/docs/app/api-reference/functions/use-search-params) hook. Since Client Components re-render on navigation, they have access to the latest query parameters.
```

Comment:
```suggestion
To access updated query parameters, you can use the Page [`searchParams`](/docs/app/api-reference/file-conventions/page#searchparams-optional) prop, or read them inside a Client Component using the [`useSearchParams`](/docs/app/api-reference/functions/use-search-params) hook. Since Client Components re-render on navigation, they have access to the latest query parameters.
```